salt/metalk8s/kubernetes/etcd/installed.sls

```diff
@@ -12,18 +12,24 @@ include:
 {#- Get the list of existing etcd member. #}
 {%- set etcd_members = pillar.metalk8s.etcd.members %}
 
-{#- Compute the initial state according to the existing list of node. #}
-{%- set state = "existing" if etcd_members else "new" %}
+{%- if pillar.is_bootstrap %}
+  {%- set state = 'new' %}
+  {%- set etcd_endpoints = {node_name: endpoint} %}
+{%- else %}
+  {%- set state = 'existing' %}
 
-{%- set etcd_endpoints = {} %}
-{#- NOTE: Filter out members with empty name as they are not started yet. #}
-{%- for member in etcd_members | selectattr('name') %}
-  {#- NOTE: Only take first peer_urls for endpoint. #}
-  {%- do etcd_endpoints.update({member['name']: member['peer_urls'][0]}) %}
-{%- endfor %}
+  {%- set etcd_endpoints = {} %}
+  {#- NOTE: Filter out members with empty name as they are not started yet. #}
+  {%- for member in etcd_members | selectattr('name') %}
+    {#- NOTE: Only take first peer_urls for endpoint. #}
+    {%- do etcd_endpoints.update({member['name']: member['peer_urls'][0]}) %}
+  {%- else %}
+    {{ raise('List of active etcd members is empty, cannot reference the existing cluster state.') }}
+  {%- endfor %}
 
-{#- Add ourselves to the endpoints. #}
-{%- do etcd_endpoints.update({node_name: endpoint}) %}
+  {#- Add ourselves to the endpoints. #}
+  {%- do etcd_endpoints.update({node_name: endpoint}) %}
```

Comment:
Not really like this approach, 
For me `state` should be `new` only if we are bootstraping AND we didn't have any etcd members.

Reply:
fixed

Reply:
Indeed, that would help in re-running the bootstrap script/orchestration after the cluster was already deployed and expanded. Not sure if there's a use-case, but no good reason for it to break everything either.